doc/deployment/on_premises.md

```diff
@@ -1,33 +1,181 @@
+- [Customizing your Pachyderm deployment for on-premises use](./deploy_custom.html) details the various options of the `pachctl deploy custom ...` command for an on-premises deployment.
+- [Single-node Pachyderm deployment](./single-node.html) is the document you should read when deploying Pachyderm for personal, low-volume usage.
+- [Registries](./docker_registries.html) takes you through on-premises, private Docker registry configuration.
+- [Ingress](./configuring_k8s_ingress.html) details the Kubernetes ingress configuration you'd need for using `pachctl` and the dashboard outside of the Kubernetes cluster
+- [Non-cloud object stores](./non-cloud-object-stores.html) discusses common configurations for on-premises object stores.
+
+Need information on a particular flavor of Kubernetes or object store?  Check out the [see also](#see-also) section.
+
+Troubleshooting a deployment? Check out [Troubleshooting Deployments](./deploy_troubleshooting.html).
+
+## Introduction
+
+Deploying Pachyderm successfully on-premises requires a few prerequisites and some planning.
+Pachyderm is built on [Kubernetes](https://kubernetes.io/).
+Before you can deploy Pachyderm, you or your Kubernetes administrator will need to perform the following actions:
+1. [Deploy Kubernetes](#deploying-kubernetes) on-premises.
+1. [Deploy a Kubernetes persistent volume](#deploying-a-persistent-volume) that Pachyderm will use to store administrative data.
+1. [Deploy an on-premises object store](#deploying-an-object-store) using a storage provider like [MinIO](https://min.io), [EMC's ECS](https://www.dellemc.com/storage/ecs/index.htm), or [SwiftStack](https://www.swiftstack.com/) to provide S3-compatible access to your on-premises storage.
+1. [Create a Pachyderm manifest](./deploy_custom.html#creating-a-pachyderm-manifest) by running the `pachctl deploy custom` command with appropriate arguments and the `--dry-run` flag to create a Kubernetes manifest for the Pachyderm deployment.
+1. [Edit the Pachyderm manifest](#deploy_custom.html#editing-a-pachyderm-manifest) for your particular Kubernetes deployment
+
+In this series of documents, we'll take you through the steps unique to Pachyderm.
+We assume you have some Kubernetes knowledge.
+We will point you to external resources for the general Kubernetes steps to give you background.
+
+## Best practices
+### Infrastructure as code
+We highly encourage you to apply the best practices used in developing software to managing the deployment process.
+1. Create scripts that automate as much of your processes as possible and keep them under version control.
+1. Keep copies of all artifacts, such as manifests, produced by those scripts and keep those under version control.
+1. Document your practices in the code and outside it.
+### Infrastructure in general
+Be sure that you design your Kubernetes infrastructure in accordance with recommended guidelines.
+Don't mix on-premises Kubernetes and cloud-based storage.
```

Comment:
why not? Just performance reasons or do we actually not support this?

Reply:
It could be done.Performance will make it unusable.